Automattic/jetpack
# AI Assistant: dispatch upgrade action when getting 429 from streaming

The app catches and sends the quota-exceeded error from the AI query endpoint (streaming). This is a good change to update the UI independently of the AI Assistant Feature endpoint. For this, on this PR:

* Add a new action setAiAssistantFeatureRequireUpgrade()
* Recduce the action updating the `hasFeature` and `requireUpgrade` meta props in the store
* Dispacth the action when the AI Assistant block gets the `429` error

Follow-up:
* Ensure dispatch the action from other AI Implementations 
* Simplify the process to show the banner in the AI Assistant block

Fixes #

## Proposed changes:

* AI Assistant: dispatch upgrade action when getting 429 from streaming


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Create an AI Assistant block instance
* Open the Jetpack AI sidebar
* Identify the usage Panel
* Open the Redux tool
* Select the `wordpress-com/plans` store
* Open the dev console
* Dispatch the action to set the site requires an upgrade

https://github.com/Automattic/jetpack/assets/77539/58c7318c-bc23-4115-9db9-b955f5f008ad

#### Testing the app

* Ensure the ai query endpoint responses `420`
  * Apply the patch D128547-code
  * Sandbox the API
* Request AI Assistant from the block
* Confirm you see the upgrade banner
* Also, you see the upgrade message in the Usage panel
* You may like to look at the Redux dev tool

https://github.com/Automattic/jetpack/assets/77539/49c7319a-5135-4cf8-9aa8-52a89e770700
